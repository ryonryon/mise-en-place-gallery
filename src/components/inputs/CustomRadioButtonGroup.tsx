import styled from "@emotion/styled";
import * as React from "react";

interface Props {
  name: string;
  labels: string[];
  display: string;
  defaultChecked: number;
  onClick: Function | null;
}

const CustomRadioButtonGroup = ({ name, labels, display, defaultChecked, onClick }: Props) => {
  const RadioGroupDiv = styled.div`
    & input[type="radio"] {
      display: none;
    }

    & > div {
      display: ${display};
      margin: 16px 0;
    }

    & label {
      color: #858585;
      position: relative;
      padding: 0 0 0 44px;
    }

    & label::after,
    & label::before {
      position: absolute;
      content: "";
      display: block;
      top: 50%;
    }

    & label::after {
      left: 15px;
      margin-top: -10px;
      width: 16px;
      height: 16px;
      border: 2px solid rgba(225, 225, 225, 0.3);
      border-radius: 50%;
    }

    & label::before {
      left: 20px;
      margin-top: -5px;
      width: 10px;
      height: 10px;
      background: rgba(225, 225, 225, 0);
      border-radius: 50%;
    }

    & label:hover::after {
      border: 2px solid rgba(225, 225, 225, 0.9);
      transition: all 0.5s ease-out;
    }

    & label:hover::before {
      background: rgba(225, 225, 225, 0.9);
      transition: all 0.5s ease-out;
    }

    & .checked::before {
      background: rgba(225, 225, 225, 0.7);
    }
  `;
  return (
    <RadioGroupDiv className="radio-group">
      {labels.map((value, index) => (
        <div key={index}>
          <label className={`${index === defaultChecked ? "checked" : ""}`}>
            <input
              type="radio"
              defaultChecked={index === defaultChecked ? true : false}
              name={name}
              value={index}
              onClick={event => (onClick ? onClick(event) : null)}
            />
            {value}
          </label>
        </div>
      ))}
    </RadioGroupDiv>
  );
};

CustomRadioButtonGroup.defaultProps = {
  name: "custom-radio-button",
  display: "block",
  defaultChecked: 0,
  onClick: null
};

export default CustomRadioButtonGroup;
