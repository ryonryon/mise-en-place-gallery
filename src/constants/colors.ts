export const primary = "#017aff";
export const secondary = "#6C757C";
export const success = "#28A744";
export const danger = "#DC3445";
export const warning = "#FFC104";
export const info = "#17A2B8";
export const light = "#F8F9FA";
export const dark = "#343A40";
export const white = "#FFFFFF";
