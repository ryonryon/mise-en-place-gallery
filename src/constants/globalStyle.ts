import { css } from "@emotion/core";
const GlobalStyle = css`
  body {
    margin: 0;
  }

  a {
    text-decoration: none;
    color: #e5e5e5;
  }

  hr {
    width: 10%;
    margin: 56px auto 32px auto;
    opacity: 0.3;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    color: #323439;
  }
`;

export default GlobalStyle;
