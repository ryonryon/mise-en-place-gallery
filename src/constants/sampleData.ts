import CardImage from "../assets/img-for-card.jpg";

export const siteNavData: [string, string[]][] = [
  ["Buttons", ["Basic Button", "Flat Button", "Icon Button", "Square Icon Button", "Material Button", "Toggle Button"]],
  [
    "Inputs",
    [
      "Basic Text Input",
      "Underline Text Input",
      "Date Text Input",
      "Password Text Input",
      "TextInput With Validation",
      "Custom Radio Buttons",
      "Custom Checkbox"
    ]
  ],
  ["Cards", ["Basic Card", "Picture Covered Card"]],
  ["Lists", ["Normal List", "Card List"]],
  ["Avatars", ["Avatar"]],
  ["Loaders", ["SpinLoader"]],
  ["Navigations", ["Side Navigation", "Top Navigation"]],
  ["Slideshows", ["Normal Slideshow"]],
  ["Composite Component", ["Searchiable List", "SignIn Form with Validation"]]
];

export const cardData = {
  title: "Lorem ipsum dolor",
  content:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Nisi porta lorem mollis aliquam.",
  cardImg: CardImage
};

export const slideshowData = [
  "https://picsum.photos/1200/1200/?image=804",
  "https://picsum.photos/1200/1200/?image=1005",
  "https://picsum.photos/1200/1200/?image=838"
];

export const radioData = ["radioData1", "radioData2", "radioData3"];

export const navigationItems: [string, string[] | null][] = [
  ["item 1", null],
  ["item 2", ["item 2 child 1", "item 2 child 2", "item 2 child 3"]],
  ["item 3", null],
  ["item 4", ["item 4 child 1", "item 4 child 2", "item 4 child 3"]]
];

export const navigationItems2 = ["item 1", "item 2", "item 3", "item 4"];
