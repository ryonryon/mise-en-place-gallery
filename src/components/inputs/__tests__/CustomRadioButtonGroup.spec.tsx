import * as React from "react";
import * as renderer from "react-test-renderer";

import { radioData } from "../../../constants/sampleData";
import CustomRadioButtonGroup from "../CustomRadioButtonGroup";

describe("CustomRadioButtonGroup test", () => {
  it("tests CustomRadioButtonGroup renders correctly ", () => {
    const customRadioButtonGroup = renderer.create(<CustomRadioButtonGroup labels={radioData} />).toJSON();
    expect(customRadioButtonGroup).toMatchSnapshot();
  });
});
