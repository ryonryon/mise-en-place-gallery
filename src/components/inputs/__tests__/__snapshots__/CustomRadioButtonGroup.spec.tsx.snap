// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRadioButtonGroup test tests CustomRadioButtonGroup renders correctly  1`] = `
<div
  className="radio-group css-1m1zyep"
>
  <div>
    <label
      className="checked"
    >
      <input
        defaultChecked={true}
        name="custom-radio-button"
        onClick={[Function]}
        type="radio"
        value={0}
      />
      radioData1
    </label>
  </div>
  <div>
    <label
      className=""
    >
      <input
        defaultChecked={false}
        name="custom-radio-button"
        onClick={[Function]}
        type="radio"
        value={1}
      />
      radioData2
    </label>
  </div>
  <div>
    <label
      className=""
    >
      <input
        defaultChecked={false}
        name="custom-radio-button"
        onClick={[Function]}
        type="radio"
        value={2}
      />
      radioData3
    </label>
  </div>
</div>
`;
